dd-java-agent/instrumentation/jdbc/src/main/java/datadog/trace/instrumentation/jdbc/SQLCommenter.java

```diff
@@ -285,6 +300,19 @@ private static int capacity(
     if (null != version) {
       len += version.length();
     }
+
+    // Add capacity for custom fields
+    if (customFields != null) {
+      for (Map.Entry<String, String> entry : customFields.entrySet()) {
+        if (entry.getKey() != null && entry.getValue() != null) {
+          // Account for URL encoding overhead (approximately 3x for worst case)
```

Comment:
It doesn't seem like the existing implementation does this for values? And it does do it for keys implicity (by calling length on the encoded version) but in practice the keys are unlikely to need much escaping... I'd ditch the multiplications? (This is just initial allocation length so there' sno correctness issue with messing this up.)